hal/interfaces/FlashHomogeneous.cpp

```diff
@@ -17,18 +17,22 @@ namespace hal
     template<class T>
     uint32_t FlashHomogeneousBase<T>::SizeOfSector(T sectorIndex) const
     {
+        assert(sectorIndex < numberOfSectors);
         return sizeOfEachSector;
     }
 
     template<class T>
     T FlashHomogeneousBase<T>::SectorOfAddress(T address) const
     {
-        return address / sizeOfEachSector;
+        auto sectorIndex = address / sizeOfEachSector;
+        assert(sectorIndex < numberOfSectors);
```

Comment:
This assertion limits the address to be below the end address, which is not consistent with the SectorOfAddress implementation of FlashHeterogenous